tests/auto_generated/SimplicialComplex_test_autogen.py

```diff
@@ -0,0 +1,114 @@
+from SimplicialComplex import *
+import random
+
+def write_cpp_headers(file):
+    cpp_headers = """
+    #include <catch2/catch_test_macros.hpp>
+    #include <wmtk/SimplicialComplex.hpp>
+    #include <wmtk/TetMesh.hpp>
+    #include <wmtk/TriMesh.hpp>
+    #include "../tools/DEBUG_TriMesh.hpp"
+    #include <igl/read_triangle_mesh.h>
+
+    using namespace wmtk;
+
+    constexpr PrimitiveType PV = PrimitiveType::Vertex;
+    constexpr PrimitiveType PE = PrimitiveType::Edge;
+    constexpr PrimitiveType PF = PrimitiveType::Face;
+    constexpr PrimitiveType PT = PrimitiveType::Tetrahedron;
+    """
+    file.write(cpp_headers)
+
+def get_test_function(mesh, test_function_name):
+    return getattr(mesh, test_function_name)
+    
+def write_tests(file, path_to_data, mesh_name, n_samples, test_function_names):
+    mesh = load_obj(path_to_data + mesh_name)
+    # Sample some simplices
+    v_samples = random.sample(list(mesh.vertices), n_samples)
+    e_samples = random.sample(list(mesh.edges), n_samples)
+    f_samples = random.sample(list(mesh.faces), n_samples)
+
+    # also sample some simplex on the boudnary of the mesh
+    v, _, _, f, _, _ = igl.read_obj(path_to_data + mesh_name)
+    bd_loop = igl.boundary_loop(f)
+    if len(bd_loop) > 0:
+        n_bd_samples = min(n_samples, len(bd_loop))
+        indices = list(range(len(bd_loop) - 1))
+        for _ in range(n_bd_samples):
+            index = random.choice(indices)  # Choose a random index
+            v_samples.append((bd_loop[index],))
+            e_samples.append(tuple(sorted([bd_loop[index], bd_loop[(index + 1) % len(bd_loop)]])))
```

Comment:
You should also test edges that are incident to the boundary, i.e. one vertex on the boundary and one on the interior. The same holds for faces.

Reply:
Also, there might be edges that are interior but both vertices are on the boundary. Any possible combination should be tested.